File: midonet-api/src/main/java/org/midonet/api/network/PortGroupPort.java
/*
 * Copyright 2014 Midokura SARL
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.midonet.api.network;

import org.midonet.api.UriResource;
import org.midonet.api.network.validation.IsValidPortId;
import org.midonet.api.ResourceUriBuilder;
import org.midonet.api.network.validation.IsValidPortGroupId;

import javax.validation.GroupSequence;
import javax.validation.constraints.NotNull;
import javax.validation.groups.Default;
import java.net.URI;
import java.util.UUID;

/**
 * DTO to represent a membership of a port in a port group.
 */
public class PortGroupPort extends UriResource {

    @NotNull
    @IsValidPortGroupId(groups = PortGroupPortCreateExtended.class)
    private UUID portGroupId;

    @NotNull
    @IsValidPortId(groups = PortGroupPortCreateExtended.class)
    private UUID portId;

    public PortGroupPort(){
    }

    public UUID getPortGroupId() {
        return portGroupId;
    }

    public void setPortGroupId(UUID portGroupId) {
        this.portGroupId = portGroupId;
    }

    public UUID getPortId() {
        return portId;
    }

    public void setPortId(UUID portId) {
        this.portId = portId;
    }

    /**
     * @return the self URI
     */
    @Override
    public URI getUri() {
        if (getBaseUri() != null && portGroupId != null && portId != null) {
            return ResourceUriBuilder.getPortGroupPort(getBaseUri(),
                    portGroupId, portId);
        } else {
            return null;
        }
    }

    public URI getPort() {
        if (getBaseUri() != null && portId != null) {
            return ResourceUriBuilder.getPort(getBaseUri(), portId);
        } else {
            return null;
        }
    }

    public URI getPortGroup() {
        if (getBaseUri() != null && portGroupId != null) {
            return ResourceUriBuilder.getPortGroup(getBaseUri(), portGroupId);
        } else {
            return null;
        }
    }

    public interface PortGroupPortCreateExtended {
    }

    @GroupSequence({ Default.class, PortGroupPortCreateExtended.class})
    public interface PortGroupPortCreateGroupSequence {
    }


}

File: python-midonetclient/src/midonetclient/port_group_port.py
# vim: tabstop=4 shiftwidth=4 softtabstop=4

# Copyright 2013 Midokura PTE LTD.
# All Rights Reserved
#
# Licensed under the Apache License, Version 2.0 (the "License"); you may
# not use this file except in compliance with the License. You may obtain
# a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
# License for the specific language governing permissions and limitations
# under the License.


from midonetclient import resource_base
from midonetclient import vendor_media_type


class PortGroupPort(resource_base.ResourceBase):

    media_type = vendor_media_type.APPLICATION_PORTGROUP_PORT_JSON

    def __init__(self, uri, dto, auth):
        super(PortGroupPort, self).__init__(uri, dto, auth)

    def get_port_group_id(self):
        return self.dto['portGroupId']

    def get_port_id(self):
        return self.dto['portId']

    def get_port_group(self):
        return self.dto['portGroup']

    def get_port(self):
        return self.dto['port']

    def port_id(self, id_):
        self.dto['portId'] = id_
        return self
